File: WEB/albiapp/app/vaadin/views/UsersView.java
package vaadin.views;


import backend.models.FlatUser;
import backend.repositories.UserRepository;
import com.vaadin.data.Binder;
import com.vaadin.data.provider.DataProvider;
import com.vaadin.navigator.View;
import com.vaadin.ui.*;
import com.vaadin.ui.renderers.ButtonRenderer;
import vaadin.components.EditorInterface;
import vaadin.components.EntityEditor;
import vaadin.components.EntityGrid;

import javax.inject.Inject;
import javax.inject.Provider;
import java.util.Arrays;

public class UsersView extends EntityGrid<FlatUser> implements View {
    private final Provider<UserEditor> userEditorProvider;

    @Inject
    public UsersView(UserRepository userRepository, Provider<UserEditor> userEditorProvider) {
        super(userRepository, FlatUser.class);
        this.userEditorProvider = userEditorProvider;
    }

    @Override
    protected void addColumns(Grid<FlatUser> grid) {
        grid.addColumn("userName");
        grid.addColumn("nickname");
        grid.addColumn("email");
        grid.addColumn("phoneNumber");
    }

    @Override
    protected EditorInterface<FlatUser> createEditor() {
        UserEditor userEditor = userEditorProvider.get();

        userEditor.init();
        return userEditor;
    }

    private static class UserEditor extends EntityEditor<FlatUser> {
        TextField userName = new TextField("User name");
        TextField name = new TextField("Name");
        TextField bankAccountNumber = new TextField("Bank account number");
        TextField phoneNumber = new TextField("Phone number");
        TextField email = new TextField("E-mail");
        TextField nickname = new TextField("Nick name");

        @Inject
        private UserEditor() {
            super(FlatUser.class);
        }

        @Override
        protected void bindFields(Binder<FlatUser> binder) {
            binder.bind(userName, "userName");
            binder.bind(name, "name");
            binder.bind(bankAccountNumber, "bankAccountNumber");
            binder.bind(phoneNumber, "phoneNumber");
            binder.bind(email, "email");
            binder.bind(nickname, "nickname");
        }

        @Override
        protected void displayFields(FormLayout formLayout) {
            Arrays.asList(userName, name, email, nickname, phoneNumber, bankAccountNumber)
                    .forEach(formLayout::addComponent);
        }
    }
}

File: DESKTOP/VP-Albi_Zrt_DESKTOP/VP-Albi_Zrt_DESKTOP/Pages/Views/UsersView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace VP_Albi_Zrt_DESKTOP.Pages.Views
{
    public class UsersView
    {
        Model.User ReferenceUser;

        public int ID                       { get { return ReferenceUser.ID; }          set {  } }
        public string Name                  { get { return ReferenceUser.Name; }        set { ReferenceUser.Name = value; } }
        public string Username              { get { return ReferenceUser.Username; }    set {  } }
        public string Password              { get { return ReferenceUser.Password; }    set { ReferenceUser.Password = value; } }
        public string Phone                 { get { return ReferenceUser.Phone; }       set { ReferenceUser.Phone = value; } }
        public string CreditCardNumber { get { return ReferenceUser.CreditCardNumber; } set { ReferenceUser.CreditCardNumber = value; } }
        public string Nickname              { get { return ReferenceUser.Nickname; }    set { ReferenceUser.Nickname = value; } }
        public string Email                 { get { return ReferenceUser.Email; }       set { ReferenceUser.Email = value; } }

        public UsersView (Model.User u)
        {
            ReferenceUser = u;
        }
    }
}
